app/Foliage/PreparePackageVersion.hs

```diff
@@ -118,12 +120,24 @@ preparePackageVersion inputDir metaFile = do
             | RevisionSpec {revisionTimestamp, revisionNumber} <- packageVersionRevisions pkgSpec
           ]
 
+  let lastDeprecation =
+        sortOn
+          (Down . fst)
+          [ (deprecationTimestamp, isDeprecated)
+            | DeprecationSpec {deprecationTimestamp, isDeprecated} <- packageVersionDeprecations pkgSpec
+          ]
+
+  let isDeprecated = case lastDeprecation of
+        ((_, x):_) -> x
+        [] -> False
+
   return
     PreparedPackageVersion
       { pkgId,
         pkgTimestamp = packageVersionTimestamp pkgSpec,
         pkgVersionSource = packageVersionSource pkgSpec,
         pkgVersionForce = packageVersionForce pkgSpec,
+        pkgVersionDeprecated = isDeprecated,
```

Comment:
Perhaps we should just slap an `Ord` instance to `RevisionSpec` and `DeprecationSpec`.

```suggestion
  -- FIXME here is where we chack that there are no "double deprecations" (i.e. two consecutive (in time) `deprecated = true` or `deprecated = false`)
  let pkgVersionDeprecationChanges =
        sortOn
          (Down . fst)
          [ (deprecationTimestamp, isDeprecated)
            | DeprecationSpec {deprecationTimestamp, isDeprecated} <- packageVersionDeprecations pkgSpec
          ]

  let pkgVersionIsDeprecated = case lastDeprecation of
        ((_, x):_) -> x
        [] -> False

  return
    PreparedPackageVersion
      { pkgId,
        pkgTimestamp = packageVersionTimestamp pkgSpec,
        pkgVersionSource = packageVersionSource pkgSpec,
        pkgVersionForce = packageVersionForce pkgSpec,
        pkgVersionDeprecationChanges,
        pkgVersionIsDeprecated,
```